src/FileServiceManager/FileService/DAL/Storages/FileSystemStorage/FileSystemStorage.cs

```diff
@@ -0,0 +1,132 @@
+﻿using FileService.Models;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.IO;
+using FileService.Helpers;
+using FileService.DAL.Storages.Options;
+using Microsoft.Extensions.Options;
+
+namespace FileService.DAL
+{
+    public class FileSystemStorage : IStorage
+    {
+        private readonly string _basePath;
+        private readonly int _fileSizeLimit;
+        private readonly FileExtension[] _allowedExtensions;
+        private readonly IFileSystemConfigurator _fileSystemConfigurator;
+
+        public FileSystemStorage(IOptions<StorageOption> options, IFileSystemConfigurator fileSystemConfigurator)
+        {
+            _basePath = Path.Combine((string.IsNullOrEmpty(options.Value.Path) ? Directory.GetCurrentDirectory() : options.Value.Path) + "\\Files\\");
+
+            _fileSizeLimit = options.Value.FileSizeLimit??int.MaxValue;
+
+            _fileSystemConfigurator = fileSystemConfigurator;
+
+            _allowedExtensions = options.Value.AllowedExtensions;
+
+        }
+        public async Task<FileMetadata> GetFileAsync(Guid fileId, int companyId)
+        {
+            var fileInfo = await _fileSystemConfigurator.GetConfigurationAsync(companyId, fileId);
+
+            if (fileInfo != null)
+            {
+                return fileInfo;
+            }
+            else
+            {
+                throw new InvalidOperationException($"File ID = {fileId} is not found for company = {companyId}");
+            }
+        }
+
+        public async IAsyncEnumerable<FileMetadata> GetAllFilesAsync(int companyId)
+        {
+            await foreach (var file in _fileSystemConfigurator.GetAllConfigurationsAsync(companyId))
+            {
+                yield return file;
+            }
+        }
+
+        public async Task<bool> RemoveFileAsync(Guid fileId, int companyId)
+        {
+            var fileInfo = await _fileSystemConfigurator.GetConfigurationAsync(companyId, fileId);
+
+            if(fileInfo != null)
+            {
+                if (File.Exists(fileInfo.FilePath))
+                {
+                    await _fileSystemConfigurator.RemoveConfigurationAsync(companyId, fileId);
+
+                    File.Delete(fileInfo.FilePath);
+
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public async Task<Guid> UploadFileAsync(IFormFile file, int companyId)
```

Comment:
Is it fine for the Service to be aware of IFormFile ?

Reply:
It is unwanted, but otherwise we need to do some work in controller (extract file extension, byte array, file name and length) and then pass information to our DAL method. Does it make sense to change? What potential dangers do you see here?